### PR TITLE
Functionality to Detect JSON in Shareable URL

### LINKED ISSUE #1330: JSON Detector Functionality
Given a shareable URL of map previously distorted and knitted using the Mapknitter lite service, a user has no functionality to re-construct the map and restore it to its previous state on the tile layer.    

**Please describe the desired behavior.**
The Mapknitter lite service should support the re-construction of maps given formatted shareable URLs. First step, which is the focus of this issue, is therefore to build a functionality to detect if a given shareable URL points to a JSON file.

### PR DESCRIPTION
**Pull request by segun-codes**
Fixes #1330  

Make sure these boxes are checked before your pull request (PR) is ready to be reviewed and merged. Thanks!

* [X] PR is descriptively titled 📑 and links the original issue above 🔗
* [x] tests pass -- look for a green checkbox ✔️ a few minutes after opening your PR -- or run tests locally with `grunt test`
* [X] code is in uniquely-named feature branch and has no merge conflicts 📁
* [ ] screenshots/GIFs are attached 📎 in case of UI updates
* [ ] @mention the original creator of the issue in a comment below for help or for a review

> We're happy to help you get this ready -- don't be afraid to ask for help, and **don't be discouraged** if your tests fail at first!

If tests do fail, click on the red `X` to learn why by reading the logs.

Please be sure you've reviewed our contribution guidelines at https://publiclab.org/contributing-to-public-lab-software 

Thanks!

### CODE REVIEW

**Comment by segun-codes** [comment]
Hi @jywarren, kindly review the JSON detector for merging, many thanks!

File: src/DistortableCollection.js [hunk 1]
```diff
@@ -199,6 +199,19 @@ L.DistortableCollection = L.FeatureGroup.extend({
     return reduce / imgs.length;
   },
 
+  isJsonDetected(currentURL) {
+    if (currentURL.includes('?json=')) {
+      startIndex = currentURL.lastIndexOf('.');
```
Comment (jywarren) [inline src/DistortableCollection.js:204]: We may have to change this if we add more url parameters.

**Review by jywarren — APPROVED** [review]
Looks great. Can you rebase this so it includes only one commit? In the future it'll help to work from a local branch other than main. 

Great job!

File: src/DistortableCollection.js (hunk 1 above)
Comment (segun-codes) [reply src/DistortableCollection.js:204]: Yes, but if we maintain the format where the shareable URLs always have `?json` at the initial part and `.json` at the ending part then there will be no need to change the code no matter the number of new parameters added to the URL.

**Comment by segun-codes** [comment]
> Looks great. Can you rebase this so it includes only one commit? In the future it'll help to work from a local branch other than main.
> 
> Great job!

Yeah, I actually rebased but for some reasons, it apparently did not work out. I will take a second look now. Many thanks!

**Comment by jywarren** [comment]
Got it. Moving to #1331!

**Pull request closed without merging**